java/core/src/java/org/apache/orc/OrcConf.java

```diff
@@ -136,6 +136,10 @@
           "being the first column, 1 being the next, and so on. ."),
   KRYO_SARG("orc.kryo.sarg", "orc.kryo.sarg", null,
       "The kryo and base64 encoded SearchArgument for predicate pushdown."),
+  KRYO_SARG_BUFFER("orc.kryo.sarg.buffer", null, 4096,
```

Comment:
How about rather than introducing two options, we just introduce KRYO_SARG_BUFFER and define a reasonable max? I'd propose that the buffer default to 8192. We can set the max to 16777216, which means in the worst case the buffer will double 11 times.